pkg/processor/aws_event_bridge.go

```diff
@@ -0,0 +1,189 @@
+	patternMap map[string]string // rules pattern to event bus mapping
+}
+
+type eventPattern struct {
+	Detail struct {
+		Subject []string `json:"subject,omitempty"`
+	} `json:"detail,omitempty"`
+}
+
+// NewAWSEventBridgeProcessor returns an AWS EventBridge processor for the given
+// stream source.
+func NewAWSEventBridgeProcessor(cfg connection.Config, source string, verbose bool) (Processor, error) {
+	eb := awsEventBridge{
+		source:     source,
+		verbose:    verbose,
+		patternMap: make(map[string]string),
+	}
+
+	var accessKey, secretKey, region, ruleARN string
+	switch cfg.Auth.Method {
+	case authMethodAWS:
+		accessKey = cfg.Auth.Secret["aws_access_key_id"]
+		secretKey = cfg.Auth.Secret["aws_secret_access_key"]
+	default:
+		return nil, errors.Errorf("unsupported authentication method for processor aws_event_bridge: %s", cfg.Auth.Method)
+	}
+
+	if cfg.Options["aws_region"] == "" {
+		return nil, errors.Errorf("config option %q must be specified", "aws_region")
+	}
+	region = cfg.Options["aws_region"]
+
+	if cfg.Options["aws_eventbridge_rule_arn"] == "" {
+		return nil, errors.Errorf("config option %q for this processor must be specified", "aws_eventbridge_rule_arn")
+	}
+	ruleARN = cfg.Options["aws_eventbridge_rule_arn"]
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region),
+		Credentials: credentials.NewStaticCredentials(
+			accessKey,
+			secretKey,
+			"", // a token will be created when the session it's used.
+		),
+	})
+	if err != nil {
+		return nil, errors.Errorf("could not create AWS session: %v", err)
+	}
+	eb.session = sess
+	ebSession := eventbridge.New(sess)
+	if ebSession == nil {
+		return nil, errors.Errorf("could not create AWS event bridge session")
+	}
+	eb.EventBridge = ebSession
+
+	rules, err := eb.ListRules(&eventbridge.ListRulesInput{})
+	if err != nil {
+		return nil, errors.Errorf("could not parse rules: %v", err)
+	}
+
+	// TODO: handle rules in non-default event bus
+	// TODO: handle NextToken() for paginated rule lists
+	for _, r := range rules.Rules {
+		log.Printf("found rule %s", *r.Arn)
+		switch {
+		case *r.Arn == ruleARN:
+			if r.EventPattern == nil {
+				return nil, errors.Errorf("rule event pattern must not be empty")
+			}
+
+			// log.Printf("found rule for vc-event-router: %s", *r.Arn)
+
+			var e eventPattern
+			err := json.Unmarshal([]byte(*r.EventPattern), &e)
+			if err != nil {
+				return nil, errors.Wrap(err, "could not parse event pattern:")
+			}
+
+			for _, s := range e.Detail.Subject {
+				log.Printf("adding event rule forwarding pattern %q to processor", s)
+				eb.patternMap[s] = *r.EventBusName
+			}
+			return &eb, nil
+		default:
+			continue
+		}
+
+	}
+	return nil, errors.Errorf("rule %s not found for configured AWS event bridge account", ruleARN)
+}
+
+// TODO: handle throttling/batching
+// TODO: break out into smaller funcs
+func (a *awsEventBridge) Process(moref types.ManagedObjectReference, baseEvent []types.BaseEvent) error {
```

Comment:
`a` -> `eventBridge` 
or
if `awsEventBridge` struct becomes `AWSEventBridge`  `a` -> `awsEventBridge`

Reply:
Yes to `a->eventBridge`. But if it's taken by struct, maybe `awsEvtBridge`? `EvntBridge`? Just `aws`?